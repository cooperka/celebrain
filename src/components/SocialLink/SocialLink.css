li.social-link {
  display: inline-block;
}

a.social-link {
  margin: .5rem;
  padding: .5rem;
  width: 4em;
  height: 4em;
  display: inline-block;
}

.social-icon {
  height: 4em;
  position: absolute;
}

.social-icon.black {
  opacity: 1;
  transition: all .2s ease;
}
a.social-link:hover .black {
  opacity: 0;
}

.social-icon.color {
  opacity: 0;
  transition: all .2s ease;
}
a.social-link:hover .color {
  opacity: 1;
}
