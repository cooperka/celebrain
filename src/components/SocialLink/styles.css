.SocialLink {
  display: inline-block;
}

.SocialLink a {
  margin: .5rem;
  padding: .5rem;
  display: inline-block;
  transition: all 0.2s ease;
}
.SocialLink a:hover {
  -webkit-filter: drop-shadow(0 3px 3px rgba(0,0,0,0.16));
  filter: drop-shadow(0 3px 3px rgba(0,0,0,0.16));
}

.social-icon {
  height: 3em;
  transition: all 0.2s ease;
}
.SocialLink a:hover .social-icon {
  -webkit-filter: drop-shadow(0 3px 3px rgba(0,0,0,0.23));
  filter: drop-shadow(0 3px 3px rgba(0,0,0,0.23));
}

.social-icon.black {
  position: absolute;
  opacity: 1;
}
.SocialLink a:hover .social-icon.black {
  opacity: 0;
  transform: scale(1.1);
}

.social-icon.color {
  opacity: 0;
}
.SocialLink a:hover .social-icon.color {
  opacity: 1;
  transform: scale(1.1);
}
