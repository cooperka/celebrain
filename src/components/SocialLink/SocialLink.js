import React, { Component, PropTypes } from 'react';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';

import './SocialLink.css';

class SocialLink extends Component {

  static propTypes = {
    name: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
    iconHover: PropTypes.string.isRequired,
  };

  state = {
    isHovering: false,
  };

  handleMouseOver() {
    this.setState({ isHovering: true });
  }

  handleMouseOut() {
    this.setState({ isHovering: false });
  }

  render() {
    const { isHovering } = this.state;
    const { name, url, icon, iconHover } = this.props;

    return (
      <li className="social-link">
        <a
          className="social-link"
          href={url}
          onMouseOver={() => this.handleMouseOver()}
          onMouseOut={() => this.handleMouseOut()}
        >
          <ReactCSSTransitionGroup
            transitionName="hover"
            transitionEnterTimeout={500}
            transitionLeaveTimeout={500}
          >
            <img
              key={isHovering}
              className="social-icon"
              src={isHovering ? iconHover : icon}
              alt={name}
            />
          </ReactCSSTransitionGroup>
        </a>
      </li>
    );
  }

}

export default SocialLink;
