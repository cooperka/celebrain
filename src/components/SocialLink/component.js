import React, { Component, PropTypes } from 'react';

import './styles.css';

class SocialLink extends Component {

  static propTypes = {
    name: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
    iconHover: PropTypes.string.isRequired,
  };

  render() {
    const { name, url, icon, iconHover } = this.props;

    return (
      <li className="social-link-item">
        <a className="social-link-anchor" href={url}>
          <img
            className={['social-icon', 'first', 'black'].join(' ')}
            src={icon}
            alt={name}
          />
          <img
            className={['social-icon', 'color'].join(' ')}
            src={iconHover}
            alt={name}
          />
        </a>
      </li>
    );
  }

}

export default SocialLink;
