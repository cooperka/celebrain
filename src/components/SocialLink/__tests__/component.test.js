// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import SocialLink from '../index';

it('renders', () => {
  const tree = renderer.create(
    <SocialLink
      name="name"
      url="url"
      icon="icon"
      iconHover="iconHover"
    />,
  );
  expect(tree.toJSON()).toMatchSnapshot();
});
