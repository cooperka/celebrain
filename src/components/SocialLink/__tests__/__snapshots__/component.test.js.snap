exports[`test renders 1`] = `
<li
  className="social-link-item">
  <a
    className="social-link-anchor"
    href="url">
    <img
      alt="name"
      className="social-icon first black"
      src="icon" />
    <img
      alt="name"
      className="social-icon color"
      src="iconHover" />
  </a>
</li>
`;
